panoptic_mapping/src/tools/evaluation_data_writer.cpp

```diff
@@ -11,19 +11,22 @@
 
 namespace panoptic_mapping {
 
+config_utilities::Factory::RegistrationRos<DataWriterBase, EvaluationDataWriter>
+    EvaluationDataWriter::registration_("evaluation");
+
+void EvaluationDataWriter::Config::checkParams() const {}
+
 void EvaluationDataWriter::Config::setupParamsAndPrinting() {
   setupParam("verbosity", &verbosity);
   setupParam("log_data_writer_config", &log_data_writer_config);
   setupParam("store_map_every_n_frames", &store_map_every_n_frames);
 }
 
-void EvaluationDataWriter::Config::checkParams() const {}
-
 EvaluationDataWriter::EvaluationDataWriter(const Config& config)
-    : LogDataWriter(config_.log_data_writer_config, false),
-      config_(config.checkValid()) {
+    : config_(config.checkValid()),
+      LogDataWriter(config.log_data_writer_config, false) {
   LOG_IF(INFO, config_.verbosity >= 1) << "\n" << config_.toString();
-  // NOTE(schmluk): Setup is invoked from the LogDataWriter
```

Comment:
The ctor of LogDataWriter just calls the setup method of the parent class. The one of the derived class has to be called to enable saving the maps.

Reply:
True, good catch! This was changed in a bit of a rush and not properly tested apparently (both data writers used to be the same class before).